src/sybil/hist_compat.go

```diff
@@ -17,9 +17,12 @@ func (hc *HistCompat) Max() int64 {
 	return hc.BasicHist.Max
 }
 
-// NewHist creates a new historgram using the same parameters.
-func (hc *HistCompat) NewHist() Histogram {
-	return hc.table.NewHist(&hc.Info)
+func (hc *HistCompat) NewHist(info *IntInfo) Histogram {
+	i := hc.Info
+	if info != nil {
```

Comment:
why pass an optional param?

Reply:
the idea here is that the params from hc would be used unless a non-nil value is passed in. We could just always accept them and have the callers always provide. That would look like hc.NewHist(hc.Info)